rfhub2/cli/keywords_importer.py

```diff
@@ -185,67 +202,78 @@ def delete_outdated_collections(
         :param collections: List of collections object
         :return: list of dictionaries with collection name and number of keywords.
         """
         loaded_collections = []
         for collection in collections:
-            coll_req = self.client.add_collection(collection)
+            coll_req = self.client.add_collection(collection[0])
             if coll_req[0] != 201:
                 print(coll_req[1]["detail"])
                 raise StopIteration
             collection_id = coll_req[1]["id"]
-            for keyword in collection["keywords"]:
-                keyword["collection_id"] = collection_id
+            for keyword in collection[1]:
+                keyword = KeywordCreate(
+                    **{**keyword.dict(), "collection_id": collection_id}
+                )
                 self.client.add_keyword(keyword)
             loaded_collections.append(
-                {"name": collection["name"], "keywords": len(collection["keywords"])}
+                {"name": collection[0].name, "keywords": len(collection[1])}
             )
             print(
-                f'{collection["name"]} library with {len(collection["keywords"])} keywords loaded.'
+                f"{collection[0].name} library with {len(collection[1])} keywords loaded."
             )
         return loaded_collections
 
-    def _serialise_libdoc(self, libdoc: Dict, path: str, keywords: Dict) -> Dict:
+    def _serialise_libdoc(self, libdoc: Dict, path: str) -> CollectionUpdate:
         """
         Serialises libdoc object to Collection object.
         :param libdoc: libdoc input object
         :param path: library path
-        :return: Collection object
+        :return: CollectionUpdate object
         """
-
         lib_dict = libdoc.__dict__
-        lib_dict["doc_format"] = lib_dict.pop("_setter__doc_format")
-        lib_dict["doc"] += self._extract_doc_from_libdoc_inits(lib_dict["inits"])
-        for key in ("_setter__keywords", "inits", "named_args"):
-            lib_dict.pop(key)
-        lib_dict["path"] = path
-        lib_dict["keywords"] = keywords
-        return lib_dict
-
-    def _serialise_keywords(self, libdoc: Dict) -> Dict:
+        return CollectionUpdate(
+            name=lib_dict["name"],
+            type=lib_dict["type"],
+            version=lib_dict["version"],
+            scope=lib_dict["scope"],
+            # named_args=lib_dict["named_args"], # we have not used this one, yet
+            path=path,
+            doc=lib_dict["doc"]
+            + self._extract_doc_from_libdoc_inits(lib_dict["inits"]),
+            doc_format=lib_dict["_setter__doc_format"],
+        )
+
+    def _serialise_keywords(self, libdoc: Dict) -> List[KeywordUpdate]:
         """
         Serialises keywords to Keyword object.
         :param :libdoc input object
-        :return: Keyword object
+        :return: KeywordUpdate object
         """
+        raw_keywords = [keyword.__dict__ for keyword in libdoc.keywords]
```

Comment:
Could you just create KeywordUpdate from keyword attributes in first list comprehension?

Reply:
Could You take a look now? I don't exaclty like the look of it now, however it's in one statement.